src/GenericTableExportBundle/objects/db/TPkgGenericTableExport.class.php

```diff
@@ -300,4 +302,22 @@ public static function GetInstanceFromSystemName($sSystemName)
 
         return $oGenericTableExport;
     }
+
+    private function getExportDirectoryPath(): string
+    {
+        return PATH_CMS_CUSTOMER_DATA.'/'.$this->sExportPath.'/';
+    }
+
+    private function createExportDirectoryIfNeeded(): void
+    {
+        $dir = $this->getExportDirectoryPath();
+
+        if (false === \is_dir($dir)) {
+            if (false === \mkdir($dir, 0777, true) && false === \is_dir($dir)) {
+
+                // TODO what kind of exception?
```

Comment:
I think we can omit the check if mkdir succeeded. If there's an error, `WriteExportToFile` will trigger a warning anyway (and we can't throw an exception from `WriteExportToFile` because of BC).

Reply:
both error handlings / todo removed.